src/Contracts/Data/Str.php

```diff
@@ -21,19 +21,20 @@ interface Str
     /**
      * Reverse the string.
      *
-     * @param string $str String to be evaluated.
+     * @param string $str      String to be evaluated.
+     * @param string $encoding Valid encoding.
      *
      * @since 3.0.0
      *
      * @return string
      */
-    public static function reverse(string $str) :string;
+    public static function reverse(string $str, $encoding = null) :string;
```

Comment:
It looks like the `$encoding` type is `string`, and I think the `$encoding` default value should be empty.

If we set the `$encoding` is `null`,the `$encoding` `@param` annotation should not only `string`.

Reply:
SO it should be mixed?

Reply:
The `param` should be `$encoding|null`.

And function name should be `public static function reverse(string $str, ?$encoding) :string;`.

This syntax is available on `php-7.1+`.

Reply:
The reference is available on [PHPDoc](https://docs.phpdoc.org/guides/types.html#multiple-types-combined).

Reply:
Ok thanks you

so this syntax `?$encoding` tell it can null means to say optional right?

Reply:
Yes. That's right.